keystorage/hashicorp/src/main/java/tech/pegasys/signers/hashicorp/HashicorpConnectionFactory.java

```diff
@@ -15,96 +15,72 @@
-        connectionParameters.getTimeoutMilliseconds().orElse(DEFAULT_TIMEOUT_MILLISECONDS));
+    return new HashicorpConnection(httpClient, connectionParameters);
   }
 
-  private void setTlsOptions(
-      final HttpClientOptions httpClientOptions, final TlsOptions tlsOptions) {
-    httpClientOptions.setSsl(true);
-    setTrustOptions(httpClientOptions, tlsOptions);
+  private HttpClient getHttpClient(ConnectionParameters connectionParameters) {
+
+    return httpClientMap.computeIfAbsent(
+        connectionParameters.getVaultURI(),
+        _key -> {
+          final HttpClient.Builder httpClientBuilder =
+              HttpClient.newBuilder()
+                  .connectTimeout(Duration.ofMillis(connectionParameters.getTimeoutMilliseconds()));
+          try {
+            if (connectionParameters.getTlsOptions().isPresent()) {
+              LOG.debug("Connection to hashicorp vault using TLS.");
+              httpClientBuilder.sslContext(
+                  getCustomSSLContext(connectionParameters.getTlsOptions().get()));
+            }
+            return httpClientBuilder.build();
+          } catch (final Exception e) {
+            throw new HashicorpException("Unable to initialise connection to hashicorp vault.", e);
+          }
+        });
   }
 
-  private void setTrustOptions(
-      final HttpClientOptions httpClientOptions, final TlsOptions tlsOptions) {
+  private SSLContext getCustomSSLContext(final TlsOptions tlsOptions)
+      throws GeneralSecurityException, IOException {
 
     validateTlsTrustStoreOptions(tlsOptions);
 
     if (tlsOptions.getTrustStoreType().isEmpty()) {
-      LOG.debug("Hashicorp server to authenticate against system CA");
-      return;
+      return SSLContext.getDefault();
     }
 
-    final TrustStoreType trustStoreType = tlsOptions.getTrustStoreType().get();
-    switch (trustStoreType) {
-      case JKS:
-        httpClientOptions.setTrustStoreOptions(
-            new JksOptions()
-                .setPath(tlsOptions.getTrustStorePath().toString())
-                .setPassword(tlsOptions.getTrustStorePassword()));
-        break;
-      case PKCS12:
-        httpClientOptions.setPfxTrustOptions(
-            new PfxOptions()
-                .setPath(tlsOptions.getTrustStorePath().toString())
-                .setPassword(tlsOptions.getTrustStorePassword()));
-        break;
-      case PEM:
-        httpClientOptions.setPemTrustOptions(
-            new PemTrustOptions().addCertPath(tlsOptions.getTrustStorePath().toString()));
-        break;
-      case WHITELIST:
-      case ALLOWLIST:
-        // Tuweni throws an NPE if the trustStorePath has no directory prefix, thus requiring
-        // the use of absolutePath.
-        httpClientOptions.setTrustOptions(
-            VertxTrustOptions.allowlistServers(
-                tlsOptions.getTrustStorePath().toAbsolutePath(), false));
-        break;
-    }
+    final SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
```

Comment:
Does Hashicorp require TLS 1.2? Not real clear to me whether we were using that before or not

Reply:
Previously this was shielded by vertx http client. Hashicorp by default supports `TLSv1.3` (with minimum defaults to `TLSv1.2`. Hence we should change our client ssl init to `TLSv1.3`.